controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -153,28 +168,148 @@ private StreamMetadataTasks(final StreamMetadataStore streamMetadataStore,
     }
 
     /**
-     * Scales stream segments.
+     * Helper method to perform scale operation against an scale request.
+     * This method posts a request in the request stream and then starts the scale operation while
+     * tracking its progress. Eventually, after scale completion, it sends a response to the caller.
      *
      * @param scope          scope.
      * @param stream         stream name.
      * @param sealedSegments segments to be sealed.
      * @param newRanges      key ranges for new segments.
      * @param scaleTimestamp scaling time stamp.
-     * @param contextOpt     optional context
+     * @param context        optional context
      * @return returns the newly created segments.
      */
-    @Task(name = "scaleStream", version = "1.0", resource = "{scope}/{stream}")
-    public CompletableFuture<ScaleResponse> scale(String scope, String stream, ArrayList<Integer> sealedSegments,
-                                                  ArrayList<AbstractMap.SimpleEntry<Double, Double>> newRanges, long scaleTimestamp,
-                                                  OperationContext contextOpt) {
-        return execute(
-                new Resource(scope, stream),
-                new Serializable[]{scope, stream, sealedSegments, newRanges, scaleTimestamp, null},
-                () -> scaleBody(scope, stream, sealedSegments, newRanges, scaleTimestamp, contextOpt));
+    public CompletableFuture<ScaleResponse> manualScale(String scope, String stream, List<Integer> sealedSegments,
```

Comment:
This operation is inherently asynchronous. Why not just post the scale event in the request stream and return some operation id to the caller?